jasmin_metrics_client/main.py

```diff
@@ -1,11 +1,189 @@
-"""
-Templated example function for jasmin-metrics-client.
-"""
+import logging
+from datetime import datetime
+from typing import Dict, List, Optional
 
+import pandas as pd
+from elastic_transport import ApiError
+from elasticsearch import Elasticsearch
+from elasticsearch_dsl import Response, Search
+from elasticsearch_dsl.response import Hit
 
-def main() -> None:
-    """
-    Example function for jasmin-metrics-client.
-    """
 
-    pass
+class MetricsClient:
+    def __init__(self, token: Optional[str]) -> None:
+        """
+        Initialize the MetricsClient with an optional token for authentication.
+
+        Args:
+            token (Optional[str]): The authentication token.
+        """
+        hosts = [f"http://es{i}.ceda.ac.uk:9200" for i in range(1, 9)]
+        headers = {"Authorization": f"Bearer {token}"}
```

Comment:
This might be the cause: {"Authorization": f"x-api-key {token}"} instead of {"Authorization": f"Bearer {token}"} if you are using the API key you shared with me and not a bearer token.